### PR TITLE
feat: peer cli

### PR DESCRIPTION
**Pull request by Karitham**
This PR adds a `peer add` command that lets us create a peer, set up the resources in cluster and get a peer config out. WIP.

This PR contains encryption code but it's unused for now, since we don't have the ability to decrypt it.

I would have liked to use <https://yeet.0x.pt/> but it has no API and the logic to encrypt the data is in JS (it uses PGP in js), so it's more work to figure out than just doing the encryption in go and decryption in js. Neither is done so for now it just has a correct encoding and decoding process.

There's a lot of code but most of it is worthless to review. I'll comment on the sections I'd like help with.

### CODE REVIEW

File: peers.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	} else {
+		json.NewEncoder(os.Stderr).Encode(peerValue)
+		return nil
+	}
+
+	oubuf := &strings.Builder{}
+	err = Format(oubuf, WireguardConfig{
+		ConfigName: name,
+		PrivateKey: keyset.String(),
+		Address:    "10.0.0.1/24", // FIXME: I know we need to build this one from prefix + index but idk where prefix is.
```
[the remark this answers is not in the document]
Comment (Karitham) [reply peers.go:149]: So WGA nodes would advertise their own status (public key + psk + address)? I think that's the right solution. Unfortunately using status for this with regular peers means we need to wait for wga to populate it before returning a proper config, which just makes the flow more complicated?

If we make wga nodes different from peers (like with the endpoint spec or just a field) then it's easy to derive your own peer from it.

**Comment by Karitham** [comment]
I'm putting this PR on hold while I figure out how to best use status to our advantage

File: peers.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (aep) [reply peers.go:149]: Only one system should calculate the ips. 
So it's either waiting for status or setting the IP explicitly

**Comment by Karitham** [comment]
```bash
$ kubectl exec wg-665df597-7wnxz -- wga peer add kar 
# kar
[Interface]
PrivateKey = pKf1SSGwQFUeuyMFSE6XzqLMn6/MRJFp/1xBTPGzebM=
Address = fdfd::17c9:c871:a525:ce9d/128

[Peer]
Endpoint = [fdfd::1]:51820
PublicKey = EkSc/78fP7uelgOvbDtbbTWbcZd6uEdpwPWqvzTLUQE=
AllowedIPs = fdfd::/16
```

**Pull request merged**